buster/chatbot.py

```diff
@@ -149,53 +156,48 @@ def prepare_prompt(
         documents_str: str = self.prepare_documents(matched_documents, max_words=self.cfg.max_words)
         return text_before_documents + documents_str + text_before_prompt + question
 
-    def get_gpt_response(self, **completion_kwargs):
+    def get_gpt_response(self, **completion_kwargs) -> Response:
         # Call the API to generate a response
         logger.info(f"querying GPT...")
         try:
-            return openai.Completion.create(**completion_kwargs)
-
+            response = openai.Completion.create(**completion_kwargs)
         except Exception as e:
             # log the error and return a generic response instead.
             logger.exception("Error connecting to OpenAI API. See traceback:")
-            response = {"choices": [{"text": "We're having trouble connecting to OpenAI right now... Try again soon!"}]}
-            return response
+            return Response("", True, "We're having trouble connecting to OpenAI right now... Try again soon!")
+
+        text = response["choices"][0]["text"]
+        return Response(text)
```

Comment:
While I agree with having the `Response(text)` being returned, it limits us to generate a single response at a time from the `openai.completion` engine. However, in practice, if we set `n=3`, we could get 3 separate completions generated simultaneously, which can be useful in certain testing scenarios (like testing for OOD questions). This would mean we need to also support lists in the `Response.text` object.

Reply:
I don't disagree, but adding support for multiple answers is outside the scope of this PR. We need to fix the architecture, as discussed, before we can really start adding behaviour.

We can change the interface when we know what we need. For example, I think we are more likely to need a list of `tuple[Response, Iterable[Source]` than we are to need a list of answer in a single Response.